flight_computer/packets.py

```diff
@@ -107,33 +175,91 @@ def __init__(self):
 class TelemetryPacket(Packet):
     """A Packet for sending telemetry data to earth"""
 
-    type_num = 2
+    type_num = TELEMETRY_PACKET_TYPE_NUM
 
-    (_names, _fmt) = Packet.initialize_fmt_and_names("./packet_schemas/telemetry.json")
+    (_fmt, _fields) = Packet.initialize_fmt_and_fields(SCHEMAS_PATH + "/telemetry.json")
 
     def __init__(self):
         super().__init__()
 
 
 """Dict mapping type nums to the packet object"""
-_packet_types = [TelemetryPacket]
+_packet_types = [GenericPacket, TelemetryPacket]
 _packet_types = {packet_type.type_num: packet_type for packet_type in _packet_types}
 
 
 def get_packet_from_raw_data(raw_data: bytes) -> Packet:
     # Construct a generic packet to check command number
     generic_packet = GenericPacket()
-    generic_packet.raw_data = raw_data[:8]
+    generic_packet.raw_data = raw_data[:2]
+    # first byte is magic number, second byte is packet type
 
     # Check magic number to make sure it's not corrupted
-    if generic_packet.data["magic"] != 12345:
+    if generic_packet.data["magic"][0] != MAGIC_NUMBER:
         raise Exception("INVALID MAGIC NUM")
 
-    type_num = generic_packet.data["type"]
+    # Check packet type to make sure its not corrupted
+    type_num = generic_packet.data["type"][0]
     if type_num not in _packet_types:
         raise Exception("INVALID PACKET TYPE")
 
-    type = _packet_types[type_num]
-    packet = type()
+    packet_type = _packet_types[type_num]
+    packet = packet_type()
     packet.raw_data = raw_data
     return packet
+
+
+def test_instantiate_generic():
```

Comment:
Test methods should be in a different file. Feel free to make a TestPacket class that inherits from Packet